### PR TITLE
Fix webpack building on Windows

### PR DESCRIPTION
**Pull request by Naouak**
You should never check paths using manually constructed paths on Node. Use `path.join` or `path.sep` to build paths.

### CODE REVIEW

**Review by ykzts — APPROVED** [review]
good catch!

**Comment by nightpool** [comment]
have you actually tested this on windows? Windows always accepts both as path seperators

**Comment by Naouak** [comment]
I have tested on Windows. It may accept both separators but it would not change anything if you check two path together.

File: config/webpack/shared.js [hunk 1]
```diff
@@ -47,14 +47,15 @@ module.exports = {
     new webpack.optimize.CommonsChunkPlugin({
       name: 'common',
       minChunks: (module, count) => {
-        if (module.resource && /node_modules\/react-intl/.test(module.resource)) {
+        const reactIntlPathRegexp = new RegExp(`node_modules\\${sep}react-intl`);
```
Comment (ykzts) [inline config/webpack/shared.js]: It seems that it will not move if there is `\\`.

**Review by ykzts — CHANGES_REQUESTED** [review]

File: config/webpack/shared.js (hunk 1 above)
Comment (Naouak) [inline config/webpack/shared.js]: the \\ is only to add a slash to escape the separator. Without it, the regexp fails.

Comment (ykzts) [inline config/webpack/shared.js]: Will not `new RegExp` escape it?

Comment (ykzts) [inline config/webpack/shared.js]: Oh, I'm sorry, it was my misunderstanding.

**Review by ykzts — APPROVED** [review]

File: config/webpack/shared.js (hunk 1 above)
Comment (Naouak) [inline config/webpack/shared.js]: I thought too but you need an actual escaping in the regexp string to avoid confusions with a regexp class(\d \w...).

**Pull request merged**